File: providers/tikhub.py
"""
TikHub 余额查询适配器
"""
from .base import ProviderSpec, SimpleHTTPProvider


def _extract(data):
    """balance 可能在 user_data / data 下或顶层"""
    scope = data.get('user_data', data.get('data', data))
    balance = scope.get('balance') if isinstance(scope, dict) else None
    if balance is None:
        raise ValueError("无法从响应中解析 balance 字段")
    return float(balance)


class TikHubProvider(SimpleHTTPProvider):
    """TikHub 服务商适配器"""

    SPEC = ProviderSpec(
        name="TikHub",
        url="https://api.tikhub.dev/api/v1/tikhub/user/get_user_info",
        headers={"accept": "application/json"},
        extract=_extract,
    )

File: internal/provider/tikhub.go
package provider

import "errors"

// Implementation note.
var tikhubSpec = Spec{
	Key:         "tikhub",
	Name:        "TikHub",
	DefaultType: "balance",
	URL:         "https://api.tikhub.dev/api/v1/tikhub/user/get_user_info",
	Headers:     map[string]string{"accept": "application/json"},
	Extract: func(data map[string]any) (float64, error) {
		// Implementation note.
		scope := data
		if v, ok := data["user_data"]; ok {
			scope = Object(v)
		} else if v, ok := data["data"]; ok {
			scope = Object(v)
		}
		balance, ok := Num(scope["balance"])
		if !ok {
			return 0, errors.New("Could not parse balance field")
		}
		return balance, nil
	},
}

func init() { RegisterSpec(tikhubSpec) }
